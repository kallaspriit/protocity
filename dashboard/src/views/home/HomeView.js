import React from 'react';
import { Link } from 'react-router-dom';
import './HomeView.scss';

const HomeView = () => (
	<div className="home-view">
		<Link to="test">test</Link>
	</div>
);

export default HomeView;
