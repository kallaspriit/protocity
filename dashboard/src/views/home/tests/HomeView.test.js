import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { HomeView } from '../HomeView';

const loadApiMock = jest.fn();
const incrementMock = jest.fn();
const decrementMock = jest.fn();

const props = {
	value: 1,
	isLoading: false,
	apiData: undefined,
	apiError: null,
	loadApi: loadApiMock,
	increment: incrementMock,
	decrement: decrementMock,
};

const wrapper = mount(
	<HomeView {...props} />,
);

describe('HomeView', () => {
	it('renders correctly', () => {
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('have correct props', () => {
		wrapper.setProps({
			value: 2,
		});

		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('increment to be called', () => {
		wrapper.find('.btn-increment').simulate('click');
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('api to be called', () => {
		wrapper.find('.btn-load-data').first().simulate('click');
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('have correct loading state', () => {
		wrapper.setProps({
			isLoading: true,
			apiData: undefined,
			apiError: null,
		});

		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('have correct success state', () => {
		wrapper.setProps({
			isLoading: false,
			apiData: 'Test api success',
			apiError: null,
		});

		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('have correct error state', () => {
		wrapper.setProps({
			isLoading: false,
			apiData: undefined,
			apiError: 'Api error',
		});

		expect(toJson(wrapper)).toMatchSnapshot();
	});
});