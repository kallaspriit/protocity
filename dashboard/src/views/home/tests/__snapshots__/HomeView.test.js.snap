exports[`HomeView api to be called 1`] = `
<HomeView
  apiError={null}
  decrement={[Function]}
  increment={[Function]}
  isLoading={false}
  loadApi={[Function]}
  value={2}>
  <div
    className="HomeView babelPlugin babelPlugin babelPlugin btn123 isVisible"
    onClick={[Function]}>
    <img
      alt="lolcat"
      src="test-file-stub" />
    <button
      className="btn btn-decrement"
      onClick={[Function]}>
      -
    </button>
    2
    <button
      className="btn btn-increment"
      onClick={[Function]}>
      +
    </button>
    <br />
    <button
      className="btn btn-load-data"
      onClick={[Function]}>
      Load data from apisdg fgsdfg
    </button>
    <button
      className="btn btn-load-data"
      onClick={[Function]}>
      Test api error
    </button>
    <p>
      API state: 
      Data not loaded
    </p>
    <br />
    <input />
  </div>
</HomeView>
`;

exports[`HomeView have correct error state 1`] = `
<HomeView
  apiError="Api error"
  decrement={[Function]}
  increment={[Function]}
  isLoading={false}
  loadApi={[Function]}
  value={2}>
  <div
    className="HomeView babelPlugin babelPlugin babelPlugin btn123 isVisible"
    onClick={[Function]}>
    <img
      alt="lolcat"
      src="test-file-stub" />
    <button
      className="btn btn-decrement"
      onClick={[Function]}>
      -
    </button>
    2
    <button
      className="btn btn-increment"
      onClick={[Function]}>
      +
    </button>
    <br />
    <button
      className="btn btn-load-data"
      onClick={[Function]}>
      Load data from apisdg fgsdfg
    </button>
    <button
      className="btn btn-load-data"
      onClick={[Function]}>
      Test api error
    </button>
    <p>
      API state: 
      Api error
    </p>
    <br />
    <input />
  </div>
</HomeView>
`;

exports[`HomeView have correct loading state 1`] = `
<HomeView
  apiError={null}
  decrement={[Function]}
  increment={[Function]}
  isLoading={true}
  loadApi={[Function]}
  value={2}>
  <div
    className="HomeView babelPlugin babelPlugin babelPlugin btn123 isVisible"
    onClick={[Function]}>
    <img
      alt="lolcat"
      src="test-file-stub" />
    <button
      className="btn btn-decrement"
      onClick={[Function]}>
      -
    </button>
    2
    <button
      className="btn btn-increment"
      onClick={[Function]}>
      +
    </button>
    <br />
    <button
      className="btn btn-load-data"
      onClick={[Function]}>
      Load data from apisdg fgsdfg
    </button>
    <button
      className="btn btn-load-data"
      onClick={[Function]}>
      Test api error
    </button>
    <p>
      API state: 
      Loading...
    </p>
    <br />
    <input />
  </div>
</HomeView>
`;

exports[`HomeView have correct props 1`] = `
<HomeView
  apiError={null}
  decrement={[Function]}
  increment={[Function]}
  isLoading={false}
  loadApi={[Function]}
  value={2}>
  <div
    className="HomeView babelPlugin babelPlugin babelPlugin btn123 isVisible"
    onClick={[Function]}>
    <img
      alt="lolcat"
      src="test-file-stub" />
    <button
      className="btn btn-decrement"
      onClick={[Function]}>
      -
    </button>
    2
    <button
      className="btn btn-increment"
      onClick={[Function]}>
      +
    </button>
    <br />
    <button
      className="btn btn-load-data"
      onClick={[Function]}>
      Load data from apisdg fgsdfg
    </button>
    <button
      className="btn btn-load-data"
      onClick={[Function]}>
      Test api error
    </button>
    <p>
      API state: 
      Data not loaded
    </p>
    <br />
    <input />
  </div>
</HomeView>
`;

exports[`HomeView have correct success state 1`] = `
<HomeView
  apiData="Test api success"
  apiError={null}
  decrement={[Function]}
  increment={[Function]}
  isLoading={false}
  loadApi={[Function]}
  value={2}>
  <div
    className="HomeView babelPlugin babelPlugin babelPlugin btn123 isVisible"
    onClick={[Function]}>
    <img
      alt="lolcat"
      src="test-file-stub" />
    <button
      className="btn btn-decrement"
      onClick={[Function]}>
      -
    </button>
    2
    <button
      className="btn btn-increment"
      onClick={[Function]}>
      +
    </button>
    <br />
    <button
      className="btn btn-load-data"
      onClick={[Function]}>
      Load data from apisdg fgsdfg
    </button>
    <button
      className="btn btn-load-data"
      onClick={[Function]}>
      Test api error
    </button>
    <p>
      API state: 
      Test api success
    </p>
    <br />
    <input />
  </div>
</HomeView>
`;

exports[`HomeView increment to be called 1`] = `
<HomeView
  apiError={null}
  decrement={[Function]}
  increment={[Function]}
  isLoading={false}
  loadApi={[Function]}
  value={2}>
  <div
    className="HomeView babelPlugin babelPlugin babelPlugin btn123 isVisible"
    onClick={[Function]}>
    <img
      alt="lolcat"
      src="test-file-stub" />
    <button
      className="btn btn-decrement"
      onClick={[Function]}>
      -
    </button>
    2
    <button
      className="btn btn-increment"
      onClick={[Function]}>
      +
    </button>
    <br />
    <button
      className="btn btn-load-data"
      onClick={[Function]}>
      Load data from apisdg fgsdfg
    </button>
    <button
      className="btn btn-load-data"
      onClick={[Function]}>
      Test api error
    </button>
    <p>
      API state: 
      Data not loaded
    </p>
    <br />
    <input />
  </div>
</HomeView>
`;

exports[`HomeView renders correctly 1`] = `
<HomeView
  apiError={null}
  decrement={[Function]}
  increment={[Function]}
  isLoading={false}
  loadApi={[Function]}
  value={1}>
  <div
    className="HomeView babelPlugin babelPlugin babelPlugin btn123 isVisible"
    onClick={[Function]}>
    <img
      alt="lolcat"
      src="test-file-stub" />
    <button
      className="btn btn-decrement"
      onClick={[Function]}>
      -
    </button>
    1
    <button
      className="btn btn-increment"
      onClick={[Function]}>
      +
    </button>
    <br />
    <button
      className="btn btn-load-data"
      onClick={[Function]}>
      Load data from apisdg fgsdfg
    </button>
    <button
      className="btn btn-load-data"
      onClick={[Function]}>
      Test api error
    </button>
    <p>
      API state: 
      Data not loaded
    </p>
    <br />
    <input />
  </div>
</HomeView>
`;
